### PR TITLE
Upgrade to Python3

### PR DESCRIPTION
**Pull request by simitt**
## Motivation/summary
This PR upgrades APM Server to use Python 3. After https://github.com/elastic/beats/pull/14798 has landed upgrading the beats framework changes python dependencies e.g. in the Makefile  from `python2` to `python3`.
In #3001 we agreed on not spending time on upgrading scripts and the system tests to `python3`. Syncing the latest beats framework showed that we would need to completely get rid of `Makefile` dependencies to avoid python version conflicts. The other option would have been to not upgrade the tests, but still upgrade the tooling, which would require handling two python versions. 

Therefore I decided to look into what's necessary to actually upgrade the python version, as changing the Makefile and ensuring everything works as expected also costs time. Turns out that only a few adoptions in the system tests were necessary (took me less than 2 hours to upgrade). 

So this PR contains
* changes to the `make update-beats` script to sync the test files
* updates to the latest beats framework
* upgrading system tests to python3: main changes were related to testing TLS configurations, please have a closer look at that when reviewing
* upgrading APM Server to python3

## Checklist

~- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~
- [x] My code follows the style guidelines of this project (run `make check-full` for static code checks and linting)
- [x] I have rebased my changes on top of the latest master branch

- [x] I have made corresponding changes to the documentation
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally with my changes

~- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~

## How to test these changes
CI passing should test a huge part of this. One-off scripts need to be tested manually. 

## Related issues
#3001

### CODE REVIEW

**Comment by Bamieh** [comment]
# [Codecov](https://codecov.io/gh/elastic/apm-server/pull/3357?src=pr&el=h1) Report
> Merging [#3357](https://codecov.io/gh/elastic/apm-server/pull/3357?src=pr&el=desc) into [master](https://codecov.io/gh/elastic/apm-server/commit/e36b645628a7eef612730e20c921dd30df647d78&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.



```diff
@@           Coverage Diff           @@
##           master    #3357   +/-   ##
=======================================
  Coverage   79.36%   79.36%           
=======================================
  Files         108      108           
  Lines        5698     5698           
=======================================
  Hits         4522     4522           
  Misses       1176     1176           
```

**Comment by simitt** [comment]
I did some manual testing for the tooling and as far as I could see everything works as expected.

**Comment by simitt** [comment]
> Only thing I'm not too sure about is the Windows CI bit. It's fine, but are we installing in the right place? How was Python being installed before?

I went along with the according [changes in beats](https://github.com/elastic/beats/pull/14798/files#diff-deb1c8ecb48863342c5d716193f0863fR68), mainly because I wouldn't know how else to solve this, and didn't think this is problematic.

The failing integration test setup is concerning, thanks for pointing that out, will look into it.

File: tests/system/apmserver.py [hunk 1]
```diff
@@ -128,7 +127,7 @@ def get_testdata_path(self, *names):
         return self._beat_path_join('testdata', *names)
 
     def get_payload(self, name):
-        with open(self.get_payload_path(name)) as f:
+        with open(self.get_payload_path(name), "r") as f:
```
[the remark this answers is not in the document]
Comment (simitt) [reply tests/system/apmserver.py]: you're right, that's a leftover.

**Comment by simitt** [comment]
Required changes for APM Integration Tests are addressed in https://github.com/elastic/apm-integration-testing/pull/766.

**Comment by simitt** [comment]
We usually don't `git rebase` on PRs, but since it makes sense to keep multiple commits here and they should be squashed I am rebasing this one.

**Pull request merged**